### PR TITLE
Adding plant-growth form dataset

### PR DESCRIPTION
**Pull request by shreyneil**
The Following PR solves https://github.com/weecology/retriever/issues/639 , Please verify and get back.

### CODE REVIEW

File: scripts/plant-growth.json [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+  "keywords": [
+    "Taxon",
+    "Plants"
+  ],
+  "name": "plant_growth_form_dataset",
+  "archived": "zip",
+  "resources": [
+    {
+      "path": "DataS1/GrowthForm_Final.txt",
+      "dialect": {
```
Comment (henrykironde) [inline scripts/plant-growth.json]: Add delimiter` \t`

**Comment by henrykironde** [comment]
For this dataset  we may want to use a custom python script. 
The file `Growthform_Scheme.txt` was compiled in a format that would require some extra effort to clean up. In fact we may get a hard time reading this file using a custom script on both python 2 and 3 but we could try. ref e9743b52

**Comment by shreyneil** [comment]
I realised that as even the encoding for Growthform_Scheme.txt is different as compared to other files.
So, do you want me to write a script for the other two files and Growthform_Scheme.txt seperately, as in , remove the Growthform_Scheme.txt from this PR and Launch another PR for Growthform_Scheme.txt

**Comment by henrykironde** [comment]
Okey create a different PR for `Growthform_Scheme.txt`.   using a custom python file. You can use this as a ref `intertidal-abund-me`. The goal is to read that file line by line and remove/trim the extra spaces. If we can get that to work, we shall add all the tables to that script.

**Comment by shreyneil** [comment]
Done with the changes,working on the Growthform_Scheme.txt part. Meanwhile , please review this one.

**Pull request closed without merging**